src/user/user.service.ts

```diff
@@ -1,12 +1,15 @@
-import { Injectable } from '@nestjs/common'
+import { Injectable, OnModuleInit } from '@nestjs/common'
+import { QueryOptions } from 'mongoose'
```

Comment:
Business logic cannot depend on `mongoose`, it should be incapsulated in DB layer.

Reply:
Fixed as:
```
// src/types/common.types.ts
import { QueryOptions as MongooseQueryOptions } from 'mongoose'
export type QueryOptions = MongooseQueryOptions
```